parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs

```diff
@@ -524,3 +534,62 @@ declExists = termExists
 before :: Branch.Hash -> Branch.Hash -> Transaction (Maybe Bool)
 before h1 h2 =
   Ops.before (Cv.causalHash1to2 h1) (Cv.causalHash1to2 h2)
+
+namesWithinPath ::
+  Maybe Path ->
+  Transaction ScopedNames
+namesWithinPath path = do
+  (termNames, typeNames) <- Ops.rootBranchNames
+  let allTerms :: [(Name, Referent.Referent)]
+      allTerms =
+        termNames <&> \(S.NamedRef {reversedSegments, ref = (ref, ct)}) ->
+          let v1ref = runIdentity $ Cv.referent2to1 (const . pure . Cv.constructorType2to1 . fromMaybe (error "Required constructor type for constructor but it was null") $ ct) ref
+           in (Name.fromReverseSegments (coerce reversedSegments), v1ref)
+  let allTypes :: [(Name, Reference.Reference)]
+      allTypes =
+        typeNames <&> \(S.NamedRef {reversedSegments, ref}) ->
+          (Name.fromReverseSegments (coerce reversedSegments), Cv.reference2to1 ref)
+  let rootTerms = Rel.fromListDomainAsc allTerms
```

Comment:
I do wonder if using `fromListDomainAsc` over `fromList` is really worth it here, since it relies on knowledge of `Name`'s `Ord` instance, which has changed at least once before (but may be settled now, I'm not sure). Anyway, if it changes again, this code will silently break.